app/src/main/java/com/example/footprnt/Map/PostDetailActivity.java

```diff
@@ -57,24 +72,73 @@ public void onClick(View v) {
+                        savedPost.saveInBackground(new SaveCallback() {
+                            @Override
+                            public void done(ParseException e) {
+                                mIsPostSaved = true;
+                                Toast.makeText(PostDetailActivity.this, "Saved Post", Toast.LENGTH_SHORT).show();
+                                mBookmark.setImageResource(R.drawable.ic_save_check_filled);
+                            }
+                        });
+                    } else {
+                        // Post already saved - unsaved
+                        ParseQuery<ParseObject> query = ParseQuery.getQuery("SavedPost");
+                        query.whereEqualTo(AppConstants.user, ParseUser.getCurrentUser()).whereEqualTo(AppConstants.post, mPost);
+                        query.findInBackground(new FindCallback<ParseObject>() {
+                            @Override
+                            public void done(List<ParseObject> objects, ParseException e) {
+                                if (e == null) {
+                                    ParseObject.deleteAllInBackground(objects, new DeleteCallback() {
+                                        @Override
+                                        public void done(ParseException e) {
+                                            mIsPostSaved = false;
+                                            mBookmark.setImageResource(R.drawable.ic_save_check);
+                                            Toast.makeText(PostDetailActivity.this, "Unsaved Post", Toast.LENGTH_SHORT).show();
+                                        }
+                                    });
+                                }
+                            }
+                        });
+                    }
+                }
+            });
+        } else {
+            mBookmark.setVisibility(View.INVISIBLE);
+        }
+    }
+
+    // Check if post is already saved
+    private void checkIfSaved() {
+        ParseQuery<ParseObject> query = ParseQuery.getQuery("SavedPost");
+        query.whereEqualTo(AppConstants.user, ParseUser.getCurrentUser()).whereEqualTo(AppConstants.post, mPost);
+        query.findInBackground(new FindCallback<ParseObject>() {
             @Override
-            public void onClick(View v) {
-                SavedPost savedPost = new SavedPost();
-                savedPost.setPost(post);
-                savedPost.setUser(ParseUser.getCurrentUser());
-                savedPost.saveInBackground(new SaveCallback() {
-                    @Override
-                    public void done(ParseException e) {
-                        Toast.makeText(PostDetailActivity.this, "Saved Post", Toast.LENGTH_LONG).show();
-                        // TODO: set tint
+            public void done(List<ParseObject> objects, ParseException e) {
+                if (e == null) {
+                    if (objects.size() > 0) {
```

Comment:
If you're just checking if it's empty or not the preferred method is

if (!objects.isEmpty()) {